test/e2e/antreapolicy_test.go

```diff
@@ -2695,6 +2694,166 @@ func testServiceAccountSelector(t *testing.T, data *TestData) {
+	}
+
+	if clusterInfo.podV6NetworkCIDR != "" {
+		ipv6Testcases := []podToAddrTestStep{
+			{
+				"x/a",
+				controlPlaneNodeIPv6(),
+				6443,
+				Dropped,
+			},
+			{
+				"x/b",
+				controlPlaneNodeIPv6(),
+				6443,
+				Connected,
+			},
+		}
+		testcases = append(testcases, ipv6Testcases...)
+	}
+	_, err := k8sUtils.CreateOrUpdateACNP(builder.Get())
+	failOnError(err, t)
+	time.Sleep(networkPolicyDelay)
+	for _, tc := range testcases {
+		log.Tracef("Probing: %s -> %s", tc.clientPod.PodName(), tc.destAddr)
+		connectivity, err := k8sUtils.ProbeAddr(tc.clientPod.Namespace(), "pod", tc.clientPod.PodName(), tc.destAddr, tc.destPort, v1.ProtocolTCP)
+		if err != nil {
+			t.Errorf("failure -- could not complete probe: %v", err)
+		}
+		if connectivity != tc.expectedConnectivity {
+			t.Errorf("failure -- wrong results for probe: Source %s/%s --> Dest %s:%d connectivity: %v, expected: %v",
+				tc.clientPod.Namespace(), tc.clientPod.PodName(), tc.destAddr, tc.destPort, connectivity, tc.expectedConnectivity)
+		}
+	}
+	// cleanup test resources
+	failOnError(k8sUtils.DeleteACNP(builder.Name), t)
+	failOnError(waitForResourceDelete("", builder.Name, resourceACNP, timeout), t)
+	time.Sleep(networkPolicyDelay)
+}
+
+func testACNPNodeSelectorIngress(t *testing.T, data *TestData) {
+	_, serverIP0, cleanupFunc := createAndWaitForPod(t, data, data.createNginxPodOnNode, "server0", nodeName(1), "x", false)
+	defer cleanupFunc()
+
+	_, serverIP1, cleanupFunc := createAndWaitForPod(t, data, data.createNginxPodOnNode, "server1", nodeName(1), "y", false)
+	defer cleanupFunc()
+
+	clientName := "agnhost-client"
+	require.NoError(t, data.createAgnhostPodOnNode(clientName, "z", controlPlaneNodeName(), true))
+	defer data.deletePodAndWait(defaultTimeout, clientName, "z")
+	_, err := data.podWaitForIPs(defaultTimeout, clientName, "z")
+	require.NoError(t, err)
+
+	clientName1 := "agnhost-client1"
+	require.NoError(t, data.createAgnhostPodOnNode(clientName1, "z", nodeName(1), true))
+	defer data.deletePodAndWait(defaultTimeout, clientName1, "z")
+	_, err = data.podWaitForIPs(defaultTimeout, clientName1, "z")
+	require.NoError(t, err)
+
+	builder := &ClusterNetworkPolicySpecBuilder{}
+	builder = builder.SetName("test-acnp-drop-ingress-from-control-plane").
+		SetPriority(1.0)
+	nodeSelector := metav1.LabelSelector{MatchLabels: map[string]string{"kubernetes.io/hostname": controlPlaneNodeName()}}
+	builder.AddNodeSelectorRule(&nodeSelector, v1.ProtocolTCP, &p80, "ingress-control-plane-drop",
+		[]ACNPAppliedToSpec{{NSSelector: map[string]string{"ns": "x"}}},
+		crdv1alpha1.RuleActionDrop, false)
+
+	testcases := []podToAddrTestStep{}
+	if clusterInfo.podV4NetworkCIDR != "" {
+		ipv4TestCases := []podToAddrTestStep{
+			{
+				Pod("z/" + clientName),
+				serverIP0.ipv4.String(),
+				80,
+				Dropped,
+			},
+			{
+				Pod("z/" + clientName),
+				serverIP1.ipv4.String(),
+				80,
+				Connected,
+			},
+			{
+				Pod("z/" + clientName1),
```

Comment:
With or without NodeSelector, this client can access this server, what does it validate?

Reply:
This case verified that the other Node to Pod traffic will not be affected by the nodeSelector rule.

Reply:
`testDefaultDenyIngressPolicy` tests probe traffic